### ISSUE TITLE
用博主的训练集训练，结果识别效果很差，跟博主已经训练好的5

### ISSUE THREAD

**Issue by liuxufenfeiya** [YCG09/chinese_ocr#77]

**Comment by liuxufenfeiya** [comment]
又试了把batch调回128，step调回博主源代码里面的数据，训练了12个小时，得出结果，跟昨天一样啊
![20180906093932433](https://user-images.githubusercontent.com/33931180/45129884-cb6fca00-b1b8-11e8-8c26-0ea2eeb7e6a4.png)是不是跟没有用预训练集keras.h5有关啊？

**Comment by liuxufenfeiya** [comment]
用了博主训练好的h5当预训练集，博主的训练数据当训练集，测试识别结果跟楼上的几乎一样，一塌糊涂，都是把不同字识别分类成某几个字的。。。训练了2000step吧，刚训练acc就0.95了，最后acc达到96%，val_acc0.95，loss0.5不到，val_loss也很低，我真是奇了怪了怎么会这样呢？

**Comment by zouj168** [comment]
4680/28184 [===>..........................] - ETA: 15702s - loss: 0.0832 - acc: 0.9895
 4681/28184 [===>..........................] - ETA: 15701s - loss: 0.0832 - acc: 0.9895


 4683/28184 [===>..........................] - ETA: 15700s - loss: 0.0832 - acc: 0.9895


 4695/28184 [===>..........................] - ETA: 15692s - loss: 0.0831 - acc: 0.9895

 4697/28184 [===>..........................] - ETA: 15690s - loss: 0.0831 - acc: 0.9895


 4727/28184 [====>.........................] - ETA: 15670s - loss: 0.0830 - acc: 0.9895
 4771/28184 [====>.........................] - ETA: 15641s - loss: 0.0827 - acc: 0.9895
28184/28184 [==============================] - 18923s - loss: 0.0775 - acc: 0.9900 - val_loss: 0.2600 - val_acc: 0.9884

和你出现情况完全一样，用数据集重新训练模型完全识别不出来。识别结果如下：




:0/task:0/device:GPU:1 with 12194 MB memory) -> physical GPU (device: 1, name: TITAN Xp, pci bus id: 0000:65:00.0, compute capability: 6.1)
Mission complete, it took 3.320s

Recognition Result:

睽哟哟哟笠哟哟哟
秀哟哟哟哟哟哟笠哟哟哟
秀睽哟
秀哟哟哟哟哟哟哟
秀睽哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟
睽哟哟哟哟哟哟哟哟哟笠哟笠哟笠哟哟哟哟哟哟哟笠哟笠哟哟哟哟
秀睽哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟笠哟哟哟笠哟哟
笠哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟哟笠哟哟哟哟哟笠哟哟哟哟哟哟笠笠笠笠哟孀笠笠哟

**Comment by zouj168** [comment]
是不是模型没导入，看看你的路径可对

**Comment by XinXinShawn** [comment]
@liuxufenfeiya 你好，我遇到了和你一样的问题，用了博主提供的360万的图片进行训练，我是在cpu上进行训练的，训练了一周的时间。识别效果也是一塌糊涂。请问你找到原因了么？

**Comment by liuxufenfeiya** [comment]
发现是模型没有导入成功，原因是博主的代码生成的是densenet_weight,而测试代码里需要的是densenet-weight，中间一个是_,一个是-，就这一点没注意，结果弄了好久。。。
另外，改完之后再试，我自己生成的2百万张图片训练完之后，并不如博主的训练集训练出来的效果好，也不知道什么情况，难道是我的训练集还是太少了么

**Comment by springtty** [comment]
这种固定格式的发票，有没有更好的文字提取方法呢

**Comment by MaxwellHan** [comment]
我也遇到了这个问题，用博主的源码训练，用的是博主的300万数据集，如果从头训练，迭代20轮也很差，而且到3、4轮之后loss就会不降反增。改小学习率的下降也没用

**Comment by YCG09** [comment]
通用模型做发票识别效果都不会太好，最好有专用数据集来训练

**Comment by springtty** [comment]
> 通用模型做发票识别效果都不会太好，最好有专用数据集来训练

谢谢博主，那是不是先找几千张发票，通过人工把需要的文本标注出来，标注后的结果再用fast rcnn从头训练识别模型，这样效果是不是更好？

**Comment by YCG09** [comment]
是的，检测和识别效果都会有所提升

**Comment by springtty** [comment]
> 是的，检测和识别效果都会有所提升

如果没有很多样本的话，那是不是可以找些空的凭证，用代码生成一些训练ctpn，是否一样有效呢

**Comment by ZouDongj** [comment]
> 发现是模型没有导入成功，原因是博主的代码生成的是densenet_weight,而测试代码里需要的是densenet-weight，中间一个是_,一个是-，就这一点没注意，结果弄了好久。。。
> 另外，改完之后再试，我自己生成的2百万张图片训练完之后，并不如博主的训练集训练出来的效果好，也不知道什么情况，难道是我的训练集还是太少了么

你好，我的测试也出现了这种问题，你说的测试代码是指model.py里面的modelPath那个路径吗？我改了那个还是不对啊……

**Comment by JJoohhnnn** [comment]
用博主训练好的.h5当做预训练模型继续用自己的样本训练，可是acc为什么还是从0还是增长，loss从一百多开始下降，训练完后模型也识别不出汉字，请问是哪里出了问题啊？

**Comment by ZouDongj** [comment]
> 用博主训练好的.h5当做预训练模型继续用自己的样本训练，可是acc为什么还是从0还是增长，loss从一百多开始下降，训练完后模型也识别不出汉字，请问是哪里出了问题啊？

看起来像是模型没加载到……识别结果是全是乱的还是识别不准确啊

**Comment by JJoohhnnn** [comment]
不是乱码，输入一张身份证图片 输出只有两个字 ，身份证里还没有出现过
train.py： 119-124行
def get_model(img_h, nclass):
    input = Input(shape=(img_h, None, 1), name='the_input')
    y_pred = densenet.dense_cnn(input, nclass)

    basemodel = Model(inputs=input, outputs=y_pred)
    basemodel.summary()

我感觉是不是 y_pred = densenet.dense_cnn(input, nclass)这里错了啊，那个.dense_cnn点进不去

**Comment by ZouDongj** [comment]
> @ZouDongj 不是乱码，输入一张身份证图片 输出只有两个字 ，身份证里还没有出现过
> train.py： 119-124行
> def get_model(img_h, nclass):
> input = Input(shape=(img_h, None, 1), name='the_input')
> y_pred = densenet.dense_cnn(input, nclass)
> 
> ```
> basemodel = Model(inputs=input, outputs=y_pred)
> basemodel.summary()
> ```
> 我感觉是不是 y_pred = densenet.dense_cnn(input, nclass)这里错了啊，那个.dense_cnn点进不去
> 预训练模型好像就没有加载进去？

你看下cptn框的区域对不对，如果框对了的话应该是识别的部分模型没加载进去，可能是路径有问题

**Comment by JJoohhnnn** [comment]
@ZouDongj  cptn框的区域没问题，识别的部分模型加载也没问题，用博主的模型检测效果挺好。我觉得是我训练的模型不对，预训练模型虽然显示加载成果，但好像还是加载不进去，不认acc也不会从0开始了。不知道该怎么改啊

**Comment by wangning7149** [comment]
请问
h5文件在哪啊

**Comment by Lanme** [comment]
请问一下，为什么你训练一个才六千多秒，用的什么设备？我跑的时间好慢啊一个就五小时了。

**Comment by hello2sys** [comment]
@liuxufenfeiya “发现是模型没有导入成功，原因是博主的代码生成的是densenet_weight,而测试代码里需要的是densenet-weight，中间一个是_,一个是-”，请问下“_”，“-”这里会有什么影响吗，我按照博主的代码用我自己的数据集重新训练生成的.h5文件直接load_weights不进去，要by_name，而且预测也跟你一样有很多乱码字符

**Comment by liuxufenfeiya** [comment]
你仔细看下载入的weights，名字是否没问题------------------ 原始邮件 ------------------
发件人: "hello2sys"<notifications@github.com>
发送时间: 2019年7月11日(星期四) 下午3:06
收件人: "YCG09/chinese_ocr"<chinese_ocr@noreply.github.com>;
抄送: "liuxufenfeiya"<343153304@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [YCG09/chinese_ocr] 用博主的训练集训练，结果识别效果很差，跟博主已经训练好的h5比差很多 (#77)



@liuxufenfeiya “发现是模型没有导入成功，原因是博主的代码生成的是densenet_weight,而测试代码里需要的是densenet-weight，中间一个是_,一个是-”，请问下“_”，“-”这里会有什么影响吗，我按照博主的代码用我自己的数据集重新训练生成的.h5文&#x4EF6;直接load_weights不进去，要by_name，而且预测也跟你一样有很多乱码字符
 
&mdash;

**Comment by v-smwang** [comment]
+1
